### ISSUE TITLE
Start porting Gym feature from Pokeroadie

### ISSUE THREAD

**Issue by samuraitruong** [NoxxDev/NecroBot#904]
Hi Guys,

I gonna to port Gym feature from PokeRoadie to our bot.

Welcome any ideal.

Truong

**Comment by bulubebek** [comment]
AFAIK, PokeRoadie automatically assign your strongest poke (based on true value) to gyms. Someone will report you if you assign Dragonite with 3k CP on a gym, for sure. ;)

**Comment by FaruqX** [comment]
what if automatically assign pokemon with cp below the strongest pokemon at gym?

**Comment by bulubebek** [comment]
yep, I think its a good solution. there is should be some options so it wont put the highest CP pokemons on gym.

btw, this is a good idea.. keep up your good job @samuraitruong

**Comment by samuraitruong** [comment]
well, up to your config  : use at your own risk :D

**Comment by samuraitruong** [comment]
to be honest, event I not using long distance snipe, I am able to get 3k CP with 100% dratini and 600 candy with human walk snipe :)

**Comment by bulubebek** [comment]
@samuraitruong wow, I mean WOW!  I should try to optimized human snipe next time. mind to share coordinates? 

by the way, I love the idea behind PokeRoadie's gym function and logic. Also their "true value" calculation logic. +1 to have it on Necro!

**Comment by mo0ojava** [comment]
Are you sure this is a good idea..? I'd recommend you keep that code private at least. Having our bot interfering with regular players like that is a surefire way of attracting unwanted attention. We should be happy that GitHub haven't removed this repo. If we want it to remain that way we should keep it under that particular radar imo.
Gym battling bot is a HUGE attention-risk.

**Comment by bulubebek** [comment]
No gym battling on Roadie.. its automatically assign pokemon to a gym with un-assigned slot(s). If pokemon died, it will attempt to revive or heal 'em.

**Comment by sevenzsu** [comment]
Would be nice to have another configuration setting for interval of collecting the rewards automatically.

**Comment by Silph-Road** [comment]
@mo0ojava I agree, no gym battles, people report players in gyms that are a high level and then you will get banned (its automated after a number of reports). There is a dedicated form for reporting and some ppl report everyone and anyone (check out reddit), Its kids who report and get ppl banned (even legit accounts) cuz they dont want to fight a level 24+ player, since they are like level ~20.

I also agree with the extra attention that might come about, Pokeease doesnt have any documentation on it either [PokeRoadie ](https://github.com/disdain13/PokeRoadie). i suggest we do not make a wiki for it or advertise it... if its implemented or maybe incorporate it in pokeease??

Yes, there should be an option to place Pokemon based on the what level the gyms is:
If the gym is level 3 then, we should be able to choose what level/CP pokemon we place in the gym, based on the level.

So setting could be like:
PlacePokemonCPLevel: 1

Level 1 (weakest pokemon CP in gym)
Level 2 (middle pokemon CP in gym)
Level 3 (strongest pokemon CP in gym)

(I dont use pokerodie, maybe they have a filter of sorts)

In general, gyms are dangerous as its not your CP pokemon.... its your badass level 30 or 28 or 27 that ppl start to hate on and then report you.

**Comment by Silph-Road** [comment]
@sevenzsu I agree 100%, without it, its pointless to put pokemons in a gym.... and after it does, it should wait before to put pokemon in gym until the timer is done and you can collect again. If you keep adding pokemon to gym and cannot collect coins, you are just risking getting reported. (This will require a setting to tell the bot to collect after it has a specific amount of gyms)

**Comment by Silph-Road** [comment]
This feature will be useful for players that are under level 20, since no one now cares about that level and assume its legit. So, a new account that uses this feature will benefit without too much risk.

Regardless, gym feature can be dangerous and attracts negative attention to the bot and the player ID.

**Comment by Silph-Road** [comment]
Check this out [https://www.reddit.com/r/pokemongodev/comments/4y0fnt/yes_you_can_get_banned_if_you_didnt_bot_or_spoof/](https://www.reddit.com/r/pokemongodev/comments/4y0fnt/yes_you_can_get_banned_if_you_didnt_bot_or_spoof/)

**Comment by samuraitruong** [comment]
yeah, it have on/off feature for people using it. will be off by default


what do you think if 1 people play game to level 30 but haven't join any team :)) haven't assign any pokemon to gym or doing any battle. If I were Niantic, I will write just a simple query to narrow down the list, then apply other condition to filter out what is bot account :))


You have your own decision to deploy what bot to GYM/maxCP/max level . so If I was you, I just random deploy < 1000 CP to gym other than no using it.


Anyway, use it at your own risk. I will use it myself



________________________________

**Comment by Silph-Road** [comment]
@samuraitruong 

> what do you think if 1 people play game to level 30 but haven't join any team :)) haven't assign any pokemon to gym or doing any battle. If I were Niantic, I will write just a simple query to narrow down the list, then apply other condition to filter out what is bot account :))

Very Interesting!! I choose a team manually when i hit level 5.. just to make it look legit, but.... never doing a battle is rather strange and very suspicious, good point !!!

**Comment by Silph-Road** [comment]
Another thing is that if you have looted over 10,000 pokestops getting to level 26+ and never visited a gym, looks very suspicious, super suspicious. 

I check out gyms just to see what kind of pokemon in it, even if i don't fight it.

Btw if you click on a gym, the app forces you to choose a team, so that's another reason why it looks more legit to have a team assigned with gym interactions.
